docs/basics/networkbehaviour.md

```diff
@@ -108,3 +108,130 @@ If you override the virtual 'OnDestroy' method it is important to alway invoke t
+:::
+
+#### Order of Operations When Dynamically Spawning:<br />
+The following provides you with an outline of the order of operations that occur during `NetworkObject` serialization when dynamically spawned.
+
+Server-Side:
+- `GameObject` with `NetworkObject` component is instantiated.
+- The `NetworkObject` is spawned. 
+  - For each associated `NetworkBehaviour` component, `NetworkBehaviour.OnNetworkSpawn` is invoked.
+- The `CreateObjectMessage` is generated 
+  - `NetworkObject` state is serialized.
+  - `NetworkVariable` state is serialized.
+  - `NetworkBehaviour.OnSynchronize` is invoked for each `NetworkBehaviour` component.
+    - If this method is not overridden then nothing is written to the serialization buffer.
+- The `CreateObjectMessage` is sent to all clients that are observers of the `NetworkObject`.
+<br />
+
+Client-Side:
+- The `CreateObjectMessage` is received
+  - `GameObject` with `NetworkObject` component is instantiated.
+  - `NetworkVariable` state is deserialized and applied.
+  - `NetworkBehaviour.OnSynchronize` is invoked for each `NetworkBehaviour` component.
+    - If this method is not overridden then nothing is read from the serialization buffer.
+- The `NetworkObject` is spawned
+  - For each associated `NetworkBehaviour` component, `NetworkBehaviour.OnNetworkSpawn` is invoked.
+
+#### Order of Operations During Full (late-join) Client Synchronization:<br />
+Server-Side:
+- The `SceneEventMessage` of type `SceneEventType.Synchronize` is created
+  - All spawned `NetworkObjects` that are visible to the client, already instantiated, and spawned are serialized.
+    - `NetworkObject` state is serialized.
+    - `NetworkVariable` state is serialized.
+    - `NetworkBehaviour.OnSynchronize` is invoked for each `NetworkBehaviour` component.
+      - If this method is not overridden then nothing is written to the serialization buffer.
+- The `SceneEventMessage` is sent to the client.
+
+Client-Side:
+- The `SceneEventMessage` of type `SceneEventType.Synchronize` is received
+- Scene information is deserialized and scenes are loaded (if not already)
+  - In-scene placed `NetworkObject`s are instantiated when a scene is loaded.
+- All `NetworkObject` oriented synchronization information is deserialized
+  - Dynamically spawned `NetworkObject`s are instantiated and state is synchronized
+  - For each `NetworkObject` instance:
+    - `NetworkVariable` state is deserialized and applied.
+    - `NetworkBehaviour.OnSynchronize` is invoked.
+      - If this method is not overridden then nothing is read from the serialization buffer.
+    - The `NetworkObject` is spawned
+      - For each associated `NetworkBehaviour` component, `NetworkBehaviour.OnNetworkSpawn` is invoked.
+
```

Comment:
Server-side:
- The server creates a `SceneEventMessage` of type `SceneEventType.Synchronize`.
  - The server serializes all `NetworkObjects` (that are visible to the client, instantiated, and spawned).
    - The server serializes the `NetworkObject` state.
    - The server serializes the `NetworkVariable` state.
    - The server invokes `NetworkBehaviour.OnSynchronize` for each `NetworkBehaviour` component.
      - If you don't override this method, the server doesn't write anything to the serialization buffer.
- The server sends `SceneEventMessage` to the client.

Client-side:
- The client receives `SceneEventMessage` of type `SceneEventType.Synchronize`.
- The client deserializes scene information and loads any unloaded scenes.
  - The client instantiates in-scene placed `NetworkObject`s when a scene loads.
- The client deserializes all `NetworkObject`-oriented synchronization information.
  - The client instantiates dynamically spawned `NetworkObject`s and synchronizes the state.
  - For each `NetworkObject` instance:
    - The client deserializes and applies the `NetworkVariable` state.
    - The client invokes `NetworkBehaviour.OnSynchronize`.
      - If you don't override this method, the client doesn't read anything from the serialization buffer.
    - The client spawns the `NetworkObject`.
      - For each associated `NetworkBehaviour` component, the client invokes `NetworkBehaviour.OnNetworkSpawn`.